tutor/src/models/task-plans/teacher/tasking.js

```diff
@@ -103,14 +100,26 @@ class TaskingPlan extends BaseModel {
     return moment(this.due_at).isAfter(Time.now);
   }
 
+  @computed get isDueAfterOpen() {
+    return moment(this.due_at).isAfter(moment(this.opens_at));
+  }
+
+  @computed get isCloseAfterDue() {
+    return moment(this.closes_at).isAfter(moment(this.due_at));
+  }
+
   @computed get isValid() {
-    return Boolean(
-      this.target_id &&
-        this.target_type &&
-        this.opens_at &&
-        this.due_at &&
-        ((this.isNew || this.dueAtChanged) ? this.isBeforeDue : true)
-    );
+    let isValid = this.target_id && this.target_type && this.opens_at && this.due_at && this.isDueAfterOpen;
+
+    if(this.isNew || this.dueAtChanged) {
+      isValid = isValid && this.isBeforeDue;
+    }
+    //event does not have a close date (visually)
+    if(!this.plan.isEvent) {
+      isValid = isValid && this.isCloseAfterDue;
+    }
+
```

Comment:
It's a little confusing to me to have isValid being checked inside the if statement How about something like this:
```js
let isValid = ...

if(isValid && (this.isNew || this.dueAtChanged)) {
  isValid =  this.isBeforeDue;
}
if(isValid && !this.plan.isEvent) {
   isValid = this.isCloseAfterDue;
}

```

Reply:
Yeah that works too. I thought about "concatenate" the booleans as we do for strings.